### PR TITLE
Add script/rubyprof to generate cachegrind callgraphs

### PR DESCRIPTION
**Pull request by fw42**
Inspired by [script/stackprof](https://github.com/jekyll/jekyll/blob/63792ad322183f159c530e9f1112b94bc333d200/script/stackprof), this adds a script to generate [ruby-prof](https://github.com/ruby-prof/ruby-prof) profiles in KCachegrind calltree format (can be opened using qcachegrind under Mac OS X).

Output looks something like this:

screen shot 2015-05-09 at 10 32 43 pm

Thoughts @parkr?

### CODE REVIEW

**Comment by parkr** [comment]
This is _super_ cool.

File: script/rubyprof [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/ruby
+
+require 'ruby-prof'
+require 'jekyll'
```
Comment (parkr) [inline script/rubyprof]: can we lock to `bundle exec` here?

File: script/rubyprof [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+RubyProf.start
+
+at_exit do
+  result = RubyProf.stop
+  printer = RubyProf::CallTreePrinter.new(result)
+  filename = "/tmp/ruby_prof_#{rand 10000}"
```
Comment (parkr) [inline script/rubyprof]: can we write to local `tmp` directory? this would vomit on windows :(

**Comment by fw42** [comment]
Updated. It's closer to stackprof now. btw I think stackprof can generate those files too. Not sure how exactly the output differs.

**Pull request merged**